Sources/TuistSupport/SecurityController.swift

```diff
@@ -0,0 +1,72 @@
+import Command
+import Mockable
+import ServiceContextModule
+
+@Mockable
+public protocol SecurityControlling {
+    func addInternetPassword(
+        accountName: String,
+        serverName: String,
+        password: String?,
+        securityProtocol: SecurityProtocol?,
+        update: Bool,
+        applications: [String]
+    ) async throws
+}
+
+public enum SecurityProtocol: String {
+    case https = "htps"
+}
+
+public struct SecurityController: SecurityControlling {
+    private let commandRunner: CommandRunning
+
+    public init(
+        commandRunner: CommandRunning = CommandRunner()
+    ) {
+        self.commandRunner = commandRunner
+    }
+
+    public func addInternetPassword(
+        accountName: String,
+        serverName: String,
+        password: String?,
+        securityProtocol: SecurityProtocol?,
+        update: Bool,
+        applications: [String]
+    ) async throws {
+        var arguments = [
+            "/usr/bin/security",
+            "add-internet-password",
+            "-a", accountName,
+            "-s", serverName,
+        ]
+
+        if let password {
+            arguments.append(
+                contentsOf: [
+                    "-w", password,
+                ]
+            )
+        }
+
+        if let securityProtocol {
+            arguments.append(
+                contentsOf: [
+                    "-r", securityProtocol.rawValue,
+                ]
+            )
+        }
+
+        if update {
+            arguments.append("-U")
+        }
+
+        arguments.append(contentsOf: applications.flatMap { ["-T", $0] })
+
+        _ = try await commandRunner.run(
+            arguments: arguments
+        )
+        .concatenatedString()
```

Comment:
What about using [`awaitCompletion`](https://github.com/tuist/Command/blob/c160f4f95235a4ec3579bc6f5cd4fd10ff5375aa/Sources/Command/AsyncThrowingStream%2BExtras.swift#L46)?

```suggestion
        try await commandRunner.run(
            arguments: arguments
        )
        . awaitCompletion()
```

Reply:
Updated the `Command` package and refactored to use this. Also added some rudimentary `SecurityControllerTests` that I forgot to add initially.

Reply:
I had to revert this suggestion because to get this, I need to bump the `Command` package and then the CI fails due to this error: https://github.com/tuist/tuist/issues/5917